kore/src/Kore/Step/RewriteStep.hs

```diff
@@ -156,8 +158,8 @@ finalizeAppliedRule renamedRule appliedConditions =
 
 finalizeRule
     ::  ( InternalVariable variable
-        , Log.WithLog Log.LogMessage unifier
-        , MonadUnify unifier
+        , MonadLog unifier
```

Comment:
Not necessary, but would it be clearer if the variable was renamed to `simplifier` since the constraint is now `MonadSimplify`? Here and in the rest of the changes in this file.